proposals/NNNN-async-stream-factory.md

```diff
@@ -0,0 +1,166 @@
+# Convenience Async[Throwing]Stream.makeStream methods
+
+* Proposal: [SE-NNNN](NNNN-async-stream-factory.md)
+* Authors: [Franz Busch](https://github.com/FranzBusch)
+* Review Manager: [Becca Royal-Gordon](https://github.com/beccadax)
+* Status: **Awaiting review**
+* Pitch: [Convenience Async[Throwing]Stream.makeStream methods](https://forums.swift.org/t/pitch-convenience-async-throwing-stream-makestream-methods/61030)
+* Implementation: [apple/swift#62968](https://github.com/apple/swift/pull/62968)
+
+<details>
+<summary><b>Revision history</b></summary>
+
+|            |                                                   |
+| ---------- | ------------------------------------------------- |
+| 2022-10-26 | Initial pitch.                                    |
+| 2023-01-12 | Switch to concrete return type                    |
+
+</details>
+
+## Introduction
+
+With [SE-0314](https://github.com/apple/swift-evolution/blob/main/proposals/0314-async-stream.md)
+we introduced `AsyncStream` and `AsyncThrowingStream` which act as a source
+`AsyncSequence` that the standard library offers.
+
+## Motivation
+
+After having used `Async[Throwing]Stream` for some time, a common usage
+is to pass the continuation and the `Async[Throwing]Stream` to different places.
+This requires escaping the `Async[Throwing]Stream.Continuation` out of 
+the closure that is passed to the initialiser.
+Escaping the continuation is slightly inconvenient since it requires a dance
+around an implicitly unwrapped optional. Furthermore, the closure implies
+that the continuation lifetime is scoped to the closure which it isn't.
+
+## Proposed solution
+
+In order to fill this gap, I propose to add a new static method `makeStream` on
+`AsyncStream` and `AsyncThrowingStream` that returns both the stream
+and the continuation.
```

Comment:
I would recommend you add a simple example of how this API would be used. This doesn't have to be anything elaborate—you can just call the new method and pass the two components off to code in different tasks. I think you can just give an example for `AsyncStream` and let people infer that it'll work the same for `AsyncThrowingStream`, or vice versa.

(To further strengthen your argument, you could add a matching example to the Motivation section showing how clumsy the closure-based API is.)